be/src/main/java/com/savethepet/model/dto/UserDto.java

```diff
@@ -0,0 +1,40 @@
+package com.savethepet.model.dto;
+
+import com.savethepet.model.entity.User;
+import lombok.Data;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+/**
+ * DTO for registration new users
+ */
+@Data
+public class UserDto {
+
+    @NotNull
+    @NotEmpty
+    private String name;
+
+    @NotNull
+    @NotEmpty
+    private String password;
+
+    @NotNull
+    @NotEmpty
+    @Email
```

Comment:
Check if `@Emai`l does all the checks described in your previous annotations? And does this work with all email services?